exp/runtime/hooks/api/v1alpha1/discovery_types.go

```diff
@@ -22,14 +22,51 @@ import (
 	"sigs.k8s.io/cluster-api/internal/runtime/catalog"
 )
 
```

Comment:
Please add the following to doc.go
```go
// +kubebuilder:object:generate=true
// +k8s:openapi-gen=true
```
(drop k8s:openapi-gen elsewhere)

Please run
```yaml
make generate-go-deepcopy-core
make generate-manifests-core
make generate-go-conversions-runtime
```